src/csharp/Grpc.Core/Internal/PlatformApis.cs

```diff
@@ -33,12 +33,17 @@ namespace Grpc.Core.Internal
     internal static class PlatformApis
     {
         const string UnityEngineApplicationClassName = "UnityEngine.Application, UnityEngine";
+        const string XamarinAndroidActivityClassName = "Android.App.Activity, Mono.Android";
```

Comment:
can you point to an article/source that suggest that this is a good way of detecting we are running on Xamarin? (thinks like that are good to doublecheck).

Reply:
Good you doublecheck ;)

Unfortunately, I can't give you a reference to a good source.

I couldn't find a better way of querying the underlying runtime from within a .netstandard1.5 library project other than how `isMono` and `isUnity` are already detected (using reflection) and chose types which most probably must exist in all versions of assemblies Mono.Android and monotouch assemblies since the beginning of time.

This should probably be changed to "Java.Lang.Object, Mono.Android" to be more similar to the "MonoTouch.Foundation.NSObject, monotouch" managed base object of the two platforms.

Is there a better way ?